### PR TITLE
[DO-NOT-MERGE] Horovod NCCL Bert

### PR DESCRIPTION
**Pull request by rondogency**
## Description ##
Just for testing horovod, keep it in separate branch

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

File: scripts/bert/launcher_mx_bert.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    distribution = {'mpi': {'enabled': True, "custom_mpi_options": "-verbose --NCCL_DEBUG=INFO"}}
+
+    estimator = MXNet(entry_point='run_pretraining_sm.py',
+                        max_run=timeout,
+                        role=role,
+                        image_uri=docker_image,
+                        source_dir='.',
+                        train_instance_count=args.instance_count,
```
Comment (ChaiBapchya) [inline scripts/bert/launcher_mx_bert.py:76]: lets update to use `instance_type` and `instance_count` since SM SDK 2.x uses these parameters.
